Thanks @reciprocum!
However this pull request does not compile:

```
[ERROR] /home/travis/build/urizev/j4gpx/src/com/urizev/gpx/GPXParser.java:[520,59] cannot find symbol
symbol:   variable trk
location: class com.urizev.gpx.GPXParser
```

--- a/src/com/urizev/gpx/GPXParser.java
+++ b/src/com/urizev/gpx/GPXParser.java
@@ -516,8 +516,8 @@ public class GPXParser {
 						.getNodeName())) {
 				    for ( IExtensionParser extensionParser : extensionParsers )
 				    {
-					Object data = extensionParser.parseTrackExtension( currentNode );
-					if (data != null) trk.addExtensionData( extensionParser.getId( ), data );
+					Object data = extensionParser.parseRouteExtension( currentNode );
+					if (data != null) rte.addExtensionData( extensionParser.getId( ), data );
 				    }
 				}
 			}
@@ -563,15 +563,10 @@ public class GPXParser {
 					trk.addTrackSegment(segment);
 				} else if (GPXConstants.EXTENSIONS_NODE.equals(currentNode
 						.getNodeName())) {
-					Iterator<IExtensionParser> it = this.extensionParsers
-							.iterator();
-					while (it.hasNext()) {
-						while (it.hasNext()) {
-							IExtensionParser parser = it.next();
-							Object data = parser
-									.parseTrackExtension(currentNode);
-							trk.addExtensionData(parser.getId(), data);
-						}
+					for ( IExtensionParser extensionParser : extensionParsers )
+					{
+						Object data = extensionParser.parseTrackExtension( currentNode );
+						if (data != null) trk.addExtensionData( extensionParser.getId( ), data );
 					}
 				}
 			}

